test/tools/fuzzTesting/createRandomTest.cpp

```diff
@@ -134,16 +135,20 @@ void dev::test::RandomCode::parseTestWithTypes(std::string& _test, std::map<std:
 				replace = test::RandomCode::randomUniIntHex(dev::u256("100000"), dev::u256("36028797018963967"));
 			else if (type == "[DESTADDRESS]")
 			{
-				Address address = _options.getRandomAddress(RandomCodeOptions::AddressType::PrecompiledOrStateOrCreate);
+				Address address = _options.getRandomAddress(RandomCodeOptions::AddressType::DestinationAccount);
 				if (address != ZeroAddress) //else transaction creation
 					replace = "0x" + toString(address);
 			}
-			else if (type == "[ADDRESS]")
-				replace = toString(_options.getRandomAddress(RandomCodeOptions::AddressType::StateAccount));
-			else if (type == "[0xADDRESS]")
+			else if (type == "[ADDRESS]") {
+				Address destAddress = _options.getRandomAddress(RandomCodeOptions::AddressType::StateAccount);
```

Comment:
so instead of making Options not const again. we better find a way to make test with accounts that are defined in options already set, rather then changing the options once we generate the account.

Reply:
The "constant options" restriction results in many useless/uninteresting random tests (e.g. send a tx with data to an empty address). Using this extra `destAddress` list makes test cases where the tx destination is an account with random code.

Reply:
@cdetrio which member of the `options` class do you want to change?  You can probably use the `mutable` keyword in the declaration of `options` class to achieve your goal.  Or,you can also separate the mutable thing from the options.

Reply:
@pirapira  its the `destinationAddressList` that needs to change, for the line `_options.addAddress(destAddress, RandomCodeOptions::AddressType::DestinationAccount);` to add an address to that list.

Reply:
@cdetrio you cannot change the command line options from inside the program.  You should copy the `destinationAddressList` to something mutable.

Reply:
These address lists are not command line options, despite being members of `_options`.  I had to change `_options` to not-const in order to call [addAddress](https://github.com/ethereum/cpp-ethereum/blob/develop/test/tools/fuzzTesting/fuzzHelper.h#L61-L62) from createRandomTest.cpp. The other calls to addAddress work because they are in [fuzzHelper.cpp](https://github.com/ethereum/cpp-ethereum/blob/develop/test/tools/fuzzTesting/fuzzHelper.cpp#L480-L493), but they're not read from the command line.

Maybe all of the address lists can be moved into something mutable? Let's figure out what exactly needs to be constant, and what doesn't.

Reply:
Then the lists should move out of `Options` class.

Reply:
Hold on, it's not `Options` but `RandomCodeOptions`.  Then just say `std::vector<dev::Address> mutable destinationAddressList;`

Reply:
why not just create a template with defined accounts. then initialize options with those accounts? 
this way options would be const and there would be no calls to non existent accounts